src/Avalonia.Controls/HotkeyManager.cs

```diff
@@ -12,21 +12,61 @@ public class HotKeyManager
 
         class HotkeyCommandWrapper : ICommand
         {
-            public HotkeyCommandWrapper(ICommandSource? control)
+            readonly WeakReference reference;
+
+            public HotkeyCommandWrapper(IControl control)
             {
-                CommandSource = control;
+                reference = new WeakReference(control);
             }
 
-            public readonly ICommandSource? CommandSource;
+            public ICommand? GetCommand()
+            {
+                if (reference.Target is { } target)
+                {
+                    if (target is ICommandSource commandSource && commandSource.Command is { } command)
+                    {
+                        return command;
```

Comment:
Am I right that Click event will stop responding to a hotkey if a command set on a button?

Reply:
> Am I right that Click event will stop responding to a hotkey if a command set on a button?

Only when CanExecute return false

> Is that how it works in WPF?

If I remember correctly it's the same behavior as WPF.